browser/ui/webui/webcompat_reporter/webcompat_reporter_dialog.cc

```diff
@@ -102,9 +111,33 @@ bool WebcompatReporterDialogDelegate::ShouldShowDialogTitle() const {
   return false;
 }
 
-void OpenWebcompatReporterDialog(content::WebContents* initiator) {
+void OpenReporterDialog(content::WebContents* initiator) {
+  bool shields_enabled = false;
+  brave_shields::mojom::FingerprintMode fp_block_mode =
+      brave_shields::mojom::FingerprintMode::STANDARD;
+  brave_shields::mojom::AdBlockMode ad_block_mode =
+      brave_shields::mojom::AdBlockMode::STANDARD;
+  brave_shields::BraveShieldsDataController* shields_data_controller =
+      brave_shields::BraveShieldsDataController::FromWebContents(initiator);
+  if (shields_data_controller != nullptr) {
+    shields_enabled = shields_data_controller->GetBraveShieldsEnabled();
+    fp_block_mode = shields_data_controller->GetFingerprintMode();
+    ad_block_mode = shields_data_controller->GetAdBlockMode();
+  }
+
+  // Remove query and fragments from reported URL.
+  GURL::Replacements replacements;
+  replacements.ClearQuery();
+  replacements.ClearRef();
+  GURL report_url =
+      initiator->GetLastCommittedURL().ReplaceComponents(replacements);
+
   base::Value::Dict params_dict;
-  params_dict.Set("siteUrl", initiator->GetLastCommittedURL().spec());
+  params_dict.Set(kSiteURLField, report_url.spec());
+  params_dict.Set(kShieldsEnabledField, shields_enabled);
+  params_dict.Set(kAdBlockSettingField, GetAdBlockModeString(ad_block_mode));
+  params_dict.Set(kFPBlockSettingField,
+                  GetFingerprintModeString(fp_block_mode));
```

Comment:
Could we define a struct for this data? I was going to let it slide since I figured it would be written directly to JSON, but that's not even the case :stuck_out_tongue: 

Especially if we can expose that struct definition in a way so it can be shared with iOS, although it doesn't have to be in this PR.

Reply:
could you maybe elaborate? this is kind of hacky, but the JSON is passed to the dialog via the webui params (which is JSON), and then those same params are passed back to the UI handler upon submission (also via JSON). I haven't found a way to pass a struct directly from the "open dialog" function to the UI handler.

Reply:
Ah, I understand now. I was hoping we could have a custom struct as a constructor argument for the delegate, but the current one that takes `base::Value` just directly sets `params_` from the inherited class. Nevermind!

Reply:
yeah it's a shame. this is a pretty cringey way to pass data, to say the least....